lib/api/core/models/repositories/userRepository.js

```diff
@@ -9,130 +11,134 @@ var
+    const
+      options = opts || {},
+      databaseOptions = options.database || {},
+      cacheOptions = options.cache || {};
 
-UserRepository.prototype.persist = function userRepositoryPersist (user, opts) {
-  var
-    options = opts || {},
-    databaseOptions = options.database || {},
-    cacheOptions = options.cache || {};
+    if (user._id === undefined || user._id === null) {
+      user._id = uuid.v4();
+    }
 
-  if (user._id === undefined || user._id === null) {
-    user._id = uuid.v4();
+    return this.persistToDatabase(user, databaseOptions)
+      .then(() => this.persistToCache(user, cacheOptions))
+      .then(() => user);
   }
 
-  return this.persistToDatabase(user, databaseOptions)
-    .then(() => this.persistToCache(user, cacheOptions))
-    .then(() => user);
-};
+  /**
+   * @returns User
+   */
+  anonymous () {
+    const user = new User();
 
-/**
- * @returns User
- */
-UserRepository.prototype.anonymous = function userRepositoryAnonymous () {
-  var user = new User();
+    return Object.assign(user, {
+      _id: '-1',
+      name: 'Anonymous',
+      profileIds: ['anonymous']
+    });
+  }
 
-  return Object.assign(user, {
-    _id: '-1',
-    name: 'Anonymous',
-    profileIds: ['anonymous']
-  });
-};
+  hydrate (user, data) {
+    let
+      source,
+      dataprofileIds;
 
-UserRepository.prototype.hydrate = function userRepositoryHydrate (user, data) {
-  var source, dataprofileIds;
+    if (!data || typeof data !== 'object') {
+      return Promise.resolve(user);
+    }
 
-  if (!data || typeof data !== 'object') {
-    return Promise.resolve(user);
-  }
+    if (data._source) {
+      source = data._source;
+      delete data._source;
+      Object.assign(data, source);
+    }
 
-  if (data._source) {
-    source = data._source;
-    delete data._source;
-    Object.assign(data, source);
-  }
+    if (data.profileIds) {
+      if (!Array.isArray(data.profileIds)) {
+        data.profileIds = [data.profileIds];
+      }
 
-  if (data.profileIds) {
-    if (!Array.isArray(data.profileIds)) {
-      data.profileIds = [data.profileIds];
+      dataprofileIds = data.profileIds;
+      delete data.profileIds;
     }
 
-    dataprofileIds = data.profileIds;
-    delete data.profileIds;
-  }
-
-  Object.assign(user, data);
-  Object.assign(user.profileIds, dataprofileIds);
+    Object.assign(user, data);
+    Object.assign(user.profileIds, dataprofileIds);
 
-  if (user._id === undefined || user._id === null) {
-    return Promise.resolve(this.anonymous());
-  }
+    if (user._id === undefined || user._id === null) {
+      return Promise.resolve(this.anonymous());
+    }
 
-  // if the user exists (have an _id) but no profile
-  // set it to default
-  if (user.profileIds.length === 0) {
-    user.profileIds = ['default'];
-  }
+    // if the user exists (have an _id) but no profile
+    // set it to default
+    if (user.profileIds.length === 0) {
+      user.profileIds = ['default'];
```

Comment:
As for #727, we should get the default Profile Ids from `restrictedProfileIds` settings.